### DISCUSSION TITLE
Guide: Making a TShock server that feels like a normal Terraria server

### DISCUSSION THREAD

**Discussion by EdoanR** [Pryaxis/TShock#2065] [Show and tell]
Hi everyone, this guide is for people that just want to play with friends with no restrictions , but still have some cool commands to use on console. Of course, you may want to do that with trusted friends :)

So on the `config.json` (file created by tshock) you should change some of these values:

```
"SpawnProtection": false,
"AnnounceSave": false,
"DisableTombstones": false,
"EnableChatAboveHeads": true,
"TileKillThreshold": 200,
"TilePlaceThreshold": 200,
"TileLiquidThreshold": 200,
"ProjectileThreshold": 200,
"HealOtherThreshold": 200,
"TilePaintThreshold": 200,
"MaxDamage": 10000,
"MaxProjDamage": 10000,
"AllowIce": true,
"AllowCutTilesAndBreakables": true,
```

You may want to change some of it, information about these you can find [here](https://tshock.readme.io/docs/config-settings).

Now for permissions stuff.
Once you started the server, type on console this:

`group addperm guest tshock.ignore.* tshock.world.modify tshock.partychat tshock.thirdperson tshock.canchat tshock.sendemoji tshock.world.movenpc tshock.npc.startinvasion tshock.npc.summonboss tshock.world.paint tshock.world.converthardmode tshock.tp.rod tshock.tp.pylon tshock.tp.wormhole tshock.npc.hurttown tshock.npc.startdd2 tshock.npc.spawnpets`

Edit: Added permission to spawn pets

All these permissions are for `guest` group, so it's not necessary login to have them, you can change it to some other group if you want.

That's it!
I hope that helped some of you. Anything that should be considered? Reply on this discussion!

Also thanks for everyone working on TShock, it's awesome.

**Comment by zaneshaw** [comment]
Thank you so much. I've had so many issues with fountains, trapdoors, etc and this guide fixed them all. Thank you!

**Reply by zaneshaw** [reply]
tshock.ignore.* will not work on SSC enabled servers as it will skip SSC backups.
EDIT: I also understand that these permissions are supposed to emulate a vanilla Terraria server so I'll change the perms my self. Anyway thanks so much for this quick guide!

**Comment by bartico6** [comment]
The recommended approach is to use the official "Vanilliafier" plugin that overrides the server configuration (that is, usergroups etc.) with values that are fitted for a relatively vanilla-ish gameplay: https://github.com/Pryaxis/Vanillafier

Of course, changing stuff in config.json isn't a part of it, but I figure it'd make sense to use it, and report things that don't work as intended so it can be improved upon for everyone's use :)

**Reply by zaneshaw** [reply]
It's kind of funny because I found out about this last night after I had already followed this guide. The next time I make a server I won't forget to use this though.

**Reply by EdoanR** [reply]
I did not know that xD
Very useful, no need to manually give permission to a group

**Comment by Anner726** [comment]
Please help me disable the bypass for the superadmin of my terraria server, since the sscconfig.json folder is useless for them with the bypass enabled.

**Reply by sgkoishi** [reply]
Simple answer: don't use superadmin. Use `owner` instead.

**Comment by Nasawi5** [comment]
When I use a consumable like Bloody Tear or Cat or Dog License, it gets consumed but does nothing. I'm using FPS.ms as a tshock server for my terraria and I already enable everything on the list, but it still does nothing.

**Reply by Kaeyve** [reply]
Same issue, goblins wont spawn

**Reply by sgkoishi** [reply]
#3079

**Comment by EdoanR** [comment]
Hey, sorry guys, it's been a long time since I made this guide (2020), not sure if still works. When I have some spare time I will test it out and try to update it.

**Comment by hakusaro** [comment]
I would also add `tshock.tp.pylon` and `tshock.world.paint` to this list.
